### PR TITLE
history.js - improving performance

### PR DESCRIPTION
**Pull request by Bournvita1998**
Signed-off-by: Mohit Kuri <mohit.kuri@research.iiit.ac.in>

### Description

Improved some performance factors present in history.js

Fixes #

Before submitting the pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by Bournvita1998** [comment]
@mauriciofauth @williamdes
Improved some performance factors in histroy.js, Should I commit further updates in history.js over here only or should I create a new PR?

cc- @ibennetch

**Comment by Bournvita1998** [comment]
@williamdes you can refer [this](https://github.com/phpmyadmin/phpmyadmin/issues/15163) for the previous discussion.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
You should resolve the conflicts before we can merge your changes.

**Comment by mauriciofauth** [comment]
Can you change your signature tag to use your full name?
We require a full name for legal compliance reasons.

For example: `Signed-off-by: Jane Smith <jane.smith@example.org>`

See [Developer's Certificate of Origin](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39)

**Review by williamdes — APPROVED** [review]
LGTM @Bournvita1998

**Comment by Bournvita1998** [comment]
> You should resolve the conflicts before we can merge your changes.

I couldn't figure out what's the exact conflict. Can you please help me here?

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
You should fix the ESLint error.
See: https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/541106844#L402

You can run ESLint with:
```
node_modules/.bin/eslint js/designer/history.js
```

**Comment by Bournvita1998** [comment]
@mauriciofauth fixed...

**Comment by Bournvita1998** [comment]
@mauriciofauth Can you please review this?
And if everything is good... should I go ahead and create a new PR "Refactoring in history.js" or should I commit those changes here in the same PR?

File: js/designer/history.js [hunk 1]
```diff
@@ -89,8 +89,9 @@ DesignerHistory.display = function (init, finit) {
     }
     // this part generates HTML code for history tab.adds delete,edit,and/or and detail features with objects.
     str = ''; // string to store Html code for history tab
-    for (i = 0; i < historyArray.length; i++) {
-        temp = historyArray[i].getTab(); // + '.' + historyArray[i].getObjNo(); for Self JOIN
+    var history_array_length = history_array.length;
```
Comment (shucon) [inline js/designer/history.js]: Use Camel Case instead of underscores to name variables.

File: js/designer/history.js [hunk 2]
```diff
@@ -89,8 +89,9 @@ DesignerHistory.display = function (init, finit) {
...
+    for (i = 0; i < history_array_length; i++) {
+        temp = history_array[i].get_tab(); // + '.' + history_array[i].get_obj_no(); for Self JOIN
```
Comment (shucon) [inline js/designer/history.js]: Don't change the variable names of existing variables.

**Review by shucon — CHANGES_REQUESTED** [review]

**Comment by shucon** [comment]
@Bournvita1998 have you even tested this PR? This code won't even work.
@williamdes @mauriciofauth please don't merge the code yet.

File: js/designer/history.js (hunk 1 above)
Comment (shucon) [reply js/designer/history.js]: There's no variable named `history_array`

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
There are still ESLint errors.

You can run ESLint with:
```
node_modules/.bin/eslint js
```
or
```
yarn run js-lint
```

**Comment by Bournvita1998** [comment]
@mauriciofauth @shucon I think while creating this PR I messed up at a lot of points(worked on different branch initially, while copying changes to the new branch got a lot of indentation issues, pulled the changes late, merge conflict, etc... ) which lead to soo many errors. I was thinking of creating a new PR, should I create one soon? I have already done work on that but now looking that how can I test that(posted the query in the mailing list regarding the same).

PS- I have considered your reviews while working this time.

**Comment by mauriciofauth** [comment]
@Bournvita1998 The majority of issues is just name changes. For example, you are using `history_array` but that variable is now named `historyArray`.

**Comment by mauriciofauth** [comment]
These conflicts were caused by some recent changes I made in the JavaScript files.

**Comment by Bournvita1998** [comment]
@mauriciofauth  ohh okay, So I guess it's okay if I resolve the issues to this PR only?

**Comment by mauriciofauth** [comment]
You can keep working on this PR.

**Comment by Bournvita1998** [comment]
@mauriciofauth while resolving ESLint error, I got these 3 errors along with other 183 warnings. Should I rename the variables or leave them as it is?

Selection_042

File: js/designer/history.js [hunk 3]
```diff
@@ -152,8 +153,9 @@ DesignerHistory.andOr = function (index) {
  *
 **/
 
-DesignerHistory.historyDelete = function (index) {
-    for (var k = 0; k < fromArray.length; k++) {
+function history_delete (index) {
```
Comment (mauriciofauth) [inline js/designer/history.js]: You have to change the signature from:
```js
function history_delete (index) {
```
to:
```js
DesignerHistory.historyDelete = function (index) {
```

File: js/designer/history.js [hunk 4]
```diff
@@ -467,8 +470,9 @@ DesignerHistory.unique = function (arrayName) {
  * @param value  value which is to be searched in the array
  */
 
-DesignerHistory.found = function (arrayName, value) {
-    for (var i = 0; i < arrayName.length; i++) {
+function found (arrayName, value) {
```
Comment (mauriciofauth) [inline js/designer/history.js]: You have to change the signature from:
```js
function found (arrayName, value) {
```
to:
```js
DesignerHistory.found = function (arrayName, value) {
```

File: js/designer/history.js [hunk 5]
```diff
@@ -482,8 +486,9 @@ DesignerHistory.found = function (arrayName, value) {
  * @params add array elements of which are pushed in
  * @params arr array in which elements are added
  */
-DesignerHistory.addArray = function (add, arr) {
-    for (var i = 0; i < add.length; i++) {
+function add_array (add, arr) {
```
Comment (mauriciofauth) [inline js/designer/history.js]: You have to change the signature from:
```js
function add_array (add, arr) {
```
to:
```js
DesignerHistory.addArray = function (add, arr) {
```

File: js/designer/history.js [hunk 6]
```diff
@@ -496,9 +501,11 @@ DesignerHistory.addArray = function (add, arr) {
  * @params arr array from which elements are removed.
  *
  */
-DesignerHistory.removeArray = function (rem, arr) {
-    for (var i = 0; i < rem.length; i++) {
-        for (var j = 0; j < arr.length; j++) {
+function remove_array (rem, arr) {
```
Comment (mauriciofauth) [inline js/designer/history.js]: You have to change the signature from:
```js
function remove_array (rem, arr) {
```
to:
```js
DesignerHistory.removeArray = function (rem, arr) {
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by Bournvita1998** [comment]
@mauriciofauth Resolved the changes suggested

**Review by MauricioFauth — APPROVED** [review]

**Comment by Bournvita1998** [comment]
@mauriciofauth Should I now move ahead to refactor history.js?

**Comment by mauriciofauth** [comment]
> Should I now move ahead to refactor history.js?

@Bournvita1998 Yes, you can open a new pull request if you want.

**Comment by Bournvita1998** [comment]
@mauriciofauth Any suggestion on the query which I asked on the developer's list and also in our personal gitter conversation?

**Pull request merged**